torch/_C/__init__.pyi.in

```diff
@@ -1269,8 +1269,16 @@ def is_grad_enabled() -> _bool: ...
 def _set_fwd_grad_enabled(enabled: _bool) -> None: ...
 def _is_fwd_grad_enabled() -> _bool: ...
 def is_inference_mode_enabled() -> _bool: ...
+@overload
```

Comment:
Here add these overloads(new signature and deprecated signature) to pass CI, in the subsequent PR(refactor autocast python APIs), we will remove the deprecated signatures to prevent the developer from using them.

Reply:
Not sure I 100% follow this. but we should keep typing and docs for deprecated APIs (with appropriate warning in doc and when it runs) until we remove them!

Reply:
OK, I will keep these deprecated signatures until they are removed in PyTorch 2.5.